tmail-backend/mailbox/encrypted/api/src/main/scala/com/linagora/tmail/encrypted/EncryptedEmailContentStore.scala

```diff
@@ -0,0 +1,66 @@
+package com.linagora.tmail.encrypted
+
+import org.apache.james.blob.api.BlobId
+import org.apache.james.mailbox.model.MessageId
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.SMono
+
+object EncryptedEmailContentStore {
+  val POSITION_NUMBER_START_AT: Int = 0
+}
+
+trait EncryptedEmailContentStore {
+  def store(messageId: MessageId, encryptedEmailContent: EncryptedEmailContent)
+
+  def delete(messageId: MessageId)
+
+  def retrieveFastView(messageId: MessageId): EncryptedEmailFastView
+
+  def retrieveDetailedView(messageId: MessageId): EncryptedEmailDetailedView
+
+  def retrieveAttachmentContent(messageId: MessageId, position: Int): BlobId
+
+  def storeReactive(messageId: MessageId, encryptedEmailContent: EncryptedEmailContent): Publisher[Unit] =
+    SMono.fromCallable(() => store(messageId, encryptedEmailContent))
+
+  def deleteReactive(messageId: MessageId): Publisher[Unit] =
+    SMono.fromCallable(() => delete(messageId))
+
+  def retrieveFastViewReactive(messageId: MessageId): Publisher[EncryptedEmailFastView] =
+    SMono.fromCallable(() => retrieveFastView(messageId))
+
+  def retrieveDetailedViewReactive(messageId: MessageId): Publisher[EncryptedEmailDetailedView] =
+    SMono.fromCallable(() => retrieveDetailedView(messageId))
+
+  def retrieveAttachmentContentReactive(messageId: MessageId, position: Int): Publisher[BlobId] =
+    SMono.fromCallable(() => retrieveAttachmentContent(messageId, position))
+}
+
+case class MessageNotFoundException(messageId: MessageId) extends Exception
+
+case class AttachmentNotFoundException(messageId: MessageId, position: Int) extends Exception
+
+object EncryptedEmailFastView {
+  def from(encryptedEmailContent: EncryptedEmailContent): EncryptedEmailFastView =
+    EncryptedEmailFastView(
+      encryptedPreview = encryptedEmailContent.encryptedPreview,
+      hasAttachment = encryptedEmailContent.hasAttachment)
+}
+
+case class EncryptedEmailFastView(encryptedPreview: String,
+                                  hasAttachment: Boolean)
+
+object EncryptedEmailDetailedView {
+  def from(encryptedEmailContent: EncryptedEmailContent): EncryptedEmailDetailedView =
+    EncryptedEmailDetailedView(
+      encryptedPreview = encryptedEmailContent.encryptedPreview,
+      encryptedHtml = encryptedEmailContent.encryptedHtml,
+      hasAttachment = encryptedEmailContent.hasAttachment,
+      encryptedAttachmentMetadata = encryptedEmailContent.encryptedAttachmentMetadata)
+}
+
+case class EncryptedEmailDetailedView(encryptedPreview: String,
+                                      encryptedHtml: String,
+                                      hasAttachment: Boolean,
+                                      encryptedAttachmentMetadata: Option[String])
```

Comment:
Reading this I wonder if we would not be better off relying on ByteSource for encryptedHtml and encryptedAttachmentMetadata...

Reply:
I don't sure about this class used to HTTP Response directly or not. If yes, I don't think ByteSource is better.

Reply:
No you are right, I thought about attachment details, sorry.